core/src/main/java/org/polypheny/db/schema/types/Typed.java

```diff
@@ -0,0 +1,26 @@
+/*
+ * Copyright 2019-2023 The Polypheny Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.polypheny.db.schema.types;
+
+import org.polypheny.db.algebra.type.AlgDataType;
+import org.polypheny.db.algebra.type.AlgDataTypeFactory;
+
+public interface Typed {
+
+    AlgDataType getRowType( AlgDataTypeFactory typeFactory );
```

Comment:
And another getRowType

Reply:
https://github.com/polypheny/Polypheny-DB/pull/464#discussion_r1518001642